models/Tree.php

```diff
@@ -251,7 +251,7 @@ public function activateNode($currNode = true)
                 }
             }
         }
-        if (!empty($this->nodeActivationErrors)) {
+        if ($this->nodeActivationErrors) {
             if ($currNode) {
                 $this->active = true;
```

Comment:
`$this->active = false` should not be done only in case of errors and therefore it might be unnecessary at this point